google-cloud-datastore/lib/google/cloud/datastore/dataset/aggregate_query_results.rb

```diff
@@ -0,0 +1,95 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module Google
+  module Cloud
+    module Datastore
+      class Dataset
+        ##
+        # # AggregateQueryResults
+        #
+        # An AggregateQueryResult object is a representation for
+        # a result of an AggregateQuery or a GqlQuery.
+        #
+        # @example
+        #   require "google/cloud/datastore"
+        #
+        #   datastore = Google::Cloud::Datastore.new
+        #
+        #   query = Google::Cloud::Datastore::Query.new
+        #   query.kind("Task").
+        #     where("done", "=", false)
+        #
+        #   Create an aggregate query
+        #   aggregate_query = query.aggregate_query
+        #                          .add_count
+        #
+        #   aggregate_query_results = dataset.run_aggregation aggregate_query
+        #   puts aggregate_query_results.get('count')
+        #
+        class AggregateQueryResults
+          ##
+          # Read timestamp the query was done on the database at.
+          #
+          # @return Google::Protobuf::Timestamp
+          attr_reader :read_time
+
+          ##
+          # Retrieves the aggregate data.
+          #
+          # @param [String] aggregate_alias The alias used
+          #   to access the aggregate value. For count, the
+          #   default value is "count".
+          #
+          # @return [Integer] The aggregate value.
+          #
+          # @example
+          #   require "google/cloud/datastore"
+          #
+          #   datastore = Google::Cloud::Datastore.new
+          #
+          #   query = Google::Cloud::Datastore::Query.new
+          #   query.kind("Task").
+          #     where("done", "=", false)
+          #
+          #   Create an aggregate query
+          #   aggregate_query = query.aggregate_query
+          #                          .add_count
+          #
+          #   aggregate_query_results = dataset.run_aggregation aggregate_query
+          #   puts aggregate_query_results.get('count')
+          def get aggregate_alias
```

Comment:
Could we make the `aggregate_alias` parameter optional if there's only one aggregate field?

Reply:
There will be multiple aggregates in the future (ex: `min`, `max`, `avg` etc). And even with just `count`, the user can specify multiple aliases (ex: `SELECT count(*) AS total, count(*) AS count FROM ...`), though there isn't much utility for them to do so. So it can't be an optional field.

Reply:
True, but I'd expect the most common use case to be a single aggregate, in which case it's annoying for a user to have to specify the correct alias name, especially if they didn't use a custom alias name so they have to know what the default is.

I'm hoping we can treat this parameter as optional if there's only one aggregate field, but required if there's more than one. That shouldn't be hard, since this is Ruby and all the checks of whether a parameter is required or not happen at runtime anyway. Just make the parameter optional overall, and the object state says there's more than one field (hence the parameter should be required) but the caller doesn't provide one then raise ArgumentError.

Reply:
I changed the `aggregate_alias` to an optional parameter. I'll do a follow up PR for Firestore's API to follow the same convention.